packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -225,6 +225,58 @@ export default util.createRule<Options, MessageIds>({
       );
     }
 
+    function isImportDeclaration(
+      definition:
+        | TSESTree.ImportDeclaration
+        | TSESTree.TSImportEqualsDeclaration,
+    ): definition is TSESTree.ImportDeclaration {
+      return definition.type === AST_NODE_TYPES.ImportDeclaration;
+    }
+
+    function isStringLiteral(
+      node: TSESTree.Node | null,
+    ): node is TSESTree.StringLiteral {
+      return (
+        !!node &&
+        node.type === AST_NODE_TYPES.Literal &&
+        typeof node.value === 'string'
+      );
+    }
+
+    function isExternalModuleDeclarationWithName(
+      scope: TSESLint.Scope.Scope,
+      name: string,
+    ): boolean {
+      return (
+        scope.type === ScopeType.tsModule &&
+        scope.block.type === AST_NODE_TYPES.TSModuleDeclaration &&
+        scope.block.id.type === AST_NODE_TYPES.Literal &&
+        scope.block.id.value === name
+      );
+    }
+
+    function isExternalDeclerationMerging(
```

Comment:
typo?

```suggestion
    function isExternalDeclarationMerging(
```
```suggestion
    function isExternalDeclerationMerging(
```

Reply:
thanks, missed that 😅 

fixed here

https://github.com/typescript-eslint/typescript-eslint/pull/3959/commits/2c97f713d7062ea969d758cf2086cf35a0cde83c